### PR TITLE
Implement lock button for appearance

### LINKED ISSUE #737: Allow locking an appearance on creation

### PR DESCRIPTION
**Pull request by YouriDK**

### CODE REVIEW

File: client/src/views/HomeView.tsx [hunk 1]
```diff
@@ -259,6 +265,11 @@ const HomeView = () => {
                 onChange={changeName}
                 value={name}
               />
+              <Tooltip title={!fixBruteAppearance ? t('lockBruteAppearance') : t('unlockBruteAppearance')}>
```
Comment (Zenoo) [inline client/src/views/HomeView.tsx]: Ternary operators should have their positive outcome first, and avoid using `!` in their main condition

File: client/src/views/HomeView.tsx [hunk 2]
```diff
@@ -259,6 +265,11 @@ const HomeView = () => {
...
+                <IconButton onClick={() => setFixBruteAppearance(!fixBruteAppearance)} size="small" sx={{ float: 'right' }}>
```
Comment (Zenoo) [inline client/src/views/HomeView.tsx]: Always use callbacks on react setters that don't depend on anything else
```suggestion
                <IconButton onClick={() => setFixBruteAppearance((prev) => !prev)} size="small" sx={{ float: 'right' }}>
```

File: client/src/views/mobile/HomeMobileView.tsx [hunk 3]
```diff
@@ -86,6 +91,11 @@ const HomeMobileView = ({
                 value={name}
                 sx={{ mx: 'auto' }}
               />
+              <Tooltip title={!fixBruteAppearance ? t('lockBruteAppearance') : t('unlockBruteAppearance')}>
+                <IconButton onClick={() => setFixBruteAppearance(!fixBruteAppearance)} size="small" sx={{ float: 'right' }}>
```
Comment (Zenoo) [inline client/src/views/mobile/HomeMobileView.tsx]: Use the setter callback instead

File: client/src/views/mobile/HomeMobileView.tsx [hunk 4]
```diff
@@ -86,6 +91,11 @@ const HomeMobileView = ({
                 value={name}
                 sx={{ mx: 'auto' }}
               />
+              <Tooltip title={!fixBruteAppearance ? t('lockBruteAppearance') : t('unlockBruteAppearance')}>
```
Comment (Zenoo) [inline client/src/views/mobile/HomeMobileView.tsx]: Invert the condition

**Review by Zenoo — CHANGES_REQUESTED** [review]
Translations are handled by Crowdin, you only need to add the translation keys in the `en` file, the others will be added later by translators

File: client/src/views/HomeView.tsx (hunk 2 above)
Comment (YouriDK) [reply client/src/views/HomeView.tsx]: I did the call back but I don't understand the (prev) => !prev) part

Comment (Zenoo) [reply client/src/views/HomeView.tsx]: It uses the previous state as reference. See https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state

File: client/src/views/HomeView.tsx [hunk 5]
```diff
@@ -222,6 +230,8 @@ const HomeView = () => {
         rightAd={rightAd}
         createBrute={createBrute}
         character={character}
+        fixBruteAppearance={fixBruteAppearance}
+        setFixBruteAppearance={changeFixBruteAppearance}
```
Comment (Zenoo) [inline client/src/views/HomeView.tsx]: This doesn't need a callback wrapping the setter, you can just use the setter.

File: client/src/views/mobile/HomeMobileView.tsx (hunk 3 above)
Comment (Zenoo) [reply client/src/views/mobile/HomeMobileView.tsx]: This needs to be the setter, and apply the same callback as the other file

**Review by Zenoo — CHANGES_REQUESTED** [review]

File: client/src/views/mobile/HomeMobileView.tsx [hunk 6]
```diff
@@ -16,6 +17,8 @@ export interface HomeMobileViewProps {
   rightAd: AdResult;
   createBrute: () => void;
   character: React.JSX.Element;
+  fixBruteAppearance: boolean;
+  setFixBruteAppearance: (params: (prev: boolean) => boolean) => void;
```
Comment (Zenoo) [inline client/src/views/mobile/HomeMobileView.tsx]: You can use the setter type instead `React.Dispatch<React.SetStateAction<string>>`

File: client/src/views/mobile/HomeMobileView.tsx [hunk 7]
```diff
@@ -86,6 +91,11 @@ const HomeMobileView = ({
                 value={name}
                 sx={{ mx: 'auto' }}
               />
+              <Tooltip title={fixBruteAppearance ? t('unlockBruteAppearance') : t('lockBruteAppearance')}>
+                <IconButton onClick={() => setFixBruteAppearance((prev: boolean) => !prev)} size="small" sx={{ float: 'right' }}>
```
Comment (Zenoo) [inline client/src/views/mobile/HomeMobileView.tsx]: You should never force type a callback parameter, as it's already typed by the setter itself

**Review by Zenoo — CHANGES_REQUESTED** [review]

**Review by Zenoo — APPROVED** [review]

**Comment by Zenoo** [comment]
Thanks for your contribution !

**Comment by YouriDK** [comment]
😬

**Pull request merged**